packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -523,7 +523,18 @@ export abstract class SharedObjectCore<TEvent extends ISharedObjectEvents = ISha
      * @returns `true` if the event had listeners, `false` otherwise.
      */
     public emit(event: EventEmitterEventType, ...args: any[]): boolean {
-        return this.callbacksHelper.measure(() => super.emit(event, ...args));
+        return this.callbacksHelper.measure(() => {
+            let result = false;
+
+            // Creating ops while handling a DDS event can lead
+            // to undefined behavior. The runtime must enforce op
```

Comment:
I'd add more concrete data here, like the fact that most DDSs (map included) would send ops in reverse order, in case of reentrancy while generating nested local edits.